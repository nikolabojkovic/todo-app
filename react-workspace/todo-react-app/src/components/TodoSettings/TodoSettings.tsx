import { useState } from "react";
import { GeneralSettings } from "./GeneralSettings/GeneralSettings";
import { PaginationSettings } from "./PaginationSettings/PaginationSettings";
import { SearchSettings } from "./SearchSettings/SearchSettings";

export function Settings() {
  
  const [paginationEnabled, setPaginationSwitch] = useState(true);
  
  return (
    <div className='App__settings'>
      <GeneralSettings {...{paginationSwitch: paginationEnabled, onPaginationSwitch: setPaginationSwitch}}></GeneralSettings>
      <SearchSettings></SearchSettings>
      <PaginationSettings isDisabled={!paginationEnabled}></PaginationSettings>
    </div>
  );
}