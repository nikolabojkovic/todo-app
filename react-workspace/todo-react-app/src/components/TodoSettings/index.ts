export * from './GeneralSettings/GeneralSettings';
export * from './PaginationSettings/PaginationSettings';
export * from './SearchSettings/SearchSettings';
export * from './TodoSettings';