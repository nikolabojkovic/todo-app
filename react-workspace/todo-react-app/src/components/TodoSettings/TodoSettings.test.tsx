import { Settings } from "./TodoSettings";

describe('', () => {
  it('should be true', () => {
    expect(true).toBe(true);
  });
});