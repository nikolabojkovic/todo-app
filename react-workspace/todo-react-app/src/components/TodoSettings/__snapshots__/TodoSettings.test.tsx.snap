// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoSettings match snapshot todo settings 1`] = `
<div
  className="App__settings"
>
  <div
    className="App__settings__group"
  >
    <label
      className="App__settings__group-label"
    >
      General
    </label>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Comfirm on change/remove
        </label>
        <div
          className="ms-auto form-switch"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id="comfirm-enabled"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Pagination
        </label>
        <div
          className="ms-auto form-switch"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id="pagination-enabled"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Infinite Scroll
        </label>
        <div
          className="ms-auto form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id="infinite-scroll-enabled"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          List container size
        </label>
        <div
          className="ms-auto dropdown"
        >
          <button
            aria-expanded={false}
            className="action-button ps-2 pe-2 dropdown-toggle btn btn-outline-secondary"
            data-testid="selected-list-size-option"
            disabled={false}
            id="dropdown-list-size-type"
            onClick={[Function]}
            type="button"
          >
             
            Dynamic
             
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="App__settings__group"
  >
    <label
      className="App__settings__group-label"
    >
      Search
    </label>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Search on key press
        </label>
        <div
          className="ms-auto form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            data-testid="search-on-keypress-switch"
            disabled={false}
            id="keyPress-switch"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="App__settings__group"
  >
    <label
      className="App__settings__group-label"
    >
      Pagination
    </label>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Pagination type
        </label>
        <div
          className="ms-auto dropdown"
        >
          <button
            aria-expanded={false}
            className="action-button ps-2 pe-2 dropdown-toggle btn btn-outline-secondary"
            data-testid="selected-pagination-type-option"
            disabled={false}
            id="dropdown-pagination-type"
            onClick={[Function]}
            type="button"
          >
             
            Classic
             
          </button>
        </div>
      </div>
    </div>
    <div
      className="App__settings__group__item"
    >
      <div
        className="hstack gap-2"
      >
        <label>
          Visible pages in navigation group
        </label>
        <div
          className="ms-auto dropdown"
        >
          <button
            aria-expanded={false}
            className="action-button ps-2 pe-2 dropdown-toggle btn btn-outline-secondary"
            data-testid="selected-visible-pages-option"
            disabled={false}
            id="dropdown-visible-pages-number"
            onClick={[Function]}
            type="button"
          >
             
            3
             
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
